### PR TITLE
Feature/replace ipl image

### PR DESCRIPTION
**Pull request by Samu24**
PR adjusts `darknet_ros` interface for the removal of `IplImage` and the C Macros.

Note: This should make `darknet_ros` ready for `noetic`

### CODE REVIEW

**Comment by Samu24** [comment]
PR depends on https://github.com/ntnu-arl/darknet/pull/1

**Comment by Samu24** [comment]
@timonh fyi

File: darknet_ros/src/YoloObjectDetector.cpp [hunk 1]
```diff
@@ -263,7 +263,7 @@ bool YoloObjectDetector::publishDetectionImage(const cv::Mat& detectionImage)
     return false;
   cv_bridge::CvImage cvImage;
   cvImage.header = headerBuff_[buffRdInd_];
-  cvImage.encoding = sensor_msgs::image_encodings::BGR8;
+  cvImage.encoding = sensor_msgs::image_encodings::RGB8;
```
Comment (marco-tranzatto) [inline darknet_ros/src/YoloObjectDetector.cpp]: is this a change of img encoding?

Comment (Samu24) [reply darknet_ros/src/YoloObjectDetector.cpp]: For the visualization yes

Comment (Samu24) [reply darknet_ros/src/YoloObjectDetector.cpp]: Apparently the visualization was not correct anymore after the changes. We can also test again on the robot to double check.

Comment (Samu24) [reply darknet_ros/src/YoloObjectDetector.cpp]: It might be related to how we create `mat_`

Comment (Samu24) [reply darknet_ros/src/YoloObjectDetector.cpp]: Ok changes in https://github.com/ntnu-arl/darknet_ros/pull/4/commits/74e43248d63e9222e3b3cc9563c19f87b5850be1 fix the requirement for the change in color coding.

**Review by marco-tranzatto — APPROVED** [review]

**Comment by HuanNguyenARL** [comment]
looks good to me

**Review by HuanNguyenARL — APPROVED** [review]
looks good to me

**Pull request merged**